.github/workflows/ci.yml

```
@@ -2,46 +2,124 @@ name: CI
 
 # Trigger the workflow on push or pull request, but only for the main branch
 on:
-  push:
-    branches:
-      - main
   pull_request:
+  push:
+    branches: [main]
 
 jobs:
   build:
-    name: ghc ${{ matrix.ghc }}
-    runs-on: ubuntu-16.04
+    name: ${{ matrix.os }} / ghc ${{ matrix.ghc }}
+    runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        cabal: ["3.0"]
+        os: [ubuntu-latest, macOS-latest, windows-latest]
+        cabal: ["3.2"]
         ghc:
           - "8.2.2"
           - "8.4.4"
           - "8.6.5"
-          - "8.8.3"
-          - "8.10.1"
+          - "8.8.4"
+          - "8.10.3"
+        exclude:
+          - os: macOS-latest
+            ghc: 8.8.4
+          - os: macOS-latest
+            ghc: 8.6.5
+          - os: macOS-latest
+            ghc: 8.4.4
+          - os: macOS-latest
+            ghc: 8.2.2
+
+          - os: windows-latest
+            ghc: 8.8.4
+          - os: windows-latest
+            ghc: 8.6.5
+          - os: windows-latest
+            ghc: 8.4.4
+          - os: windows-latest
+            ghc: 8.2.2
 
     steps:
     - uses: actions/checkout@v2
       if: github.event.action == 'opened' || github.event.action == 'synchronize' || github.event.ref == 'refs/heads/main'
 
-    - uses: actions/setup-haskell@v1
+    - uses: haskell/actions/setup@v1
+      id: setup-haskell-cabal
       name: Setup Haskell
       with:
         ghc-version: ${{ matrix.ghc }}
         cabal-version: ${{ matrix.cabal }}
 
-    - uses: actions/cache@v1
+    - name: Configure
+      run: |
+        cabal configure --enable-tests --enable-benchmarks --test-show-details=direct
```

I just noticed that this command is not perfect in my blog post. If tests won't work, this change will be required as well...
```suggestion
        cabal configure --enable-tests --enable-benchmarks --test-show-details=direct --write-ghc-environment-files=always
```